src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +24,31 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+
+    private static final int MAX_SIZE = 5;
+
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        Set<Character> chars = new TreeSet<>();
+
+        try (FileReader fileReader = new FileReader(fileName);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)) {
+            int character = bufferedReader.read();
+            while (character != -1) {
+                if (Character.isLetter(character)) {
+                    chars.add((char) Character.toLowerCase(character));
+                }
+                character = bufferedReader.read();
+            }
+        } catch (IOException exception) {
+            exception.printStackTrace();
+            throw new FileNotFoundException("File " + fileName + " is not found");
+        }
+
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < Math.min(MAX_SIZE, chars.size()); i++) {
+            result.append(chars.toArray()[i]);
```

Comment:
we are transforming the set to array in each iteration
try making this part less "expensive"